java/src/main/native/src/TableJni.cpp

```diff
@@ -1482,21 +1584,18 @@ JNIEXPORT jlong JNICALL Java_ai_rapids_cudf_Table_readJSONFromDataSource(
     cudf::jni::native_jstringArray n_col_names(env, col_names);
     cudf::jni::native_jintArray n_types(env, j_types);
     cudf::jni::native_jintArray n_scales(env, j_scales);
+    cudf::jni::native_jintArray n_children(env, j_num_children);
     if (n_types.is_null() != n_scales.is_null()) {
       JNI_THROW_NEW(env, "java/lang/IllegalArgumentException", "types and scales must match null",
                     0);
     }
-    std::vector<cudf::data_type> data_types;
-    if (!n_types.is_null()) {
-      if (n_types.size() != n_scales.size()) {
-        JNI_THROW_NEW(env, "java/lang/IllegalArgumentException", "types and scales must match size",
-                      0);
-      }
-      data_types.reserve(n_types.size());
-      std::transform(n_types.begin(), n_types.end(), n_scales.begin(),
-                     std::back_inserter(data_types), [](auto const &type, auto const &scale) {
-                       return cudf::data_type{static_cast<cudf::type_id>(type), scale};
-                     });
+    if (n_types.is_null() != n_col_names.is_null()) {
+      JNI_THROW_NEW(env, "java/lang/IllegalArgumentException", "types and names must match null",
```

Comment:
```suggestion
      JNI_THROW_NEW(env, cudf::jni::ILLEGAL_ARG_CLASS, "types and names must match null",
```

Applies to the many other uses of this raw string value below.